script/auth.js

```diff
@@ -0,0 +1,165 @@
+        isEmailValidate = false;
+        hideError(emailInput, "email-empty"); // 두 가지 오류 메세지가 동시에 나타나지 않도록 함
+        showError(emailInput, "email-invalidate");
+    } else {
+        isEmailValidate = true;
+        hideError(emailInput, "email-empty");
+        hideError(emailInput, "email-invalidate");
+    }
+    submitButtonInvalidate();
+}
+
+if (emailInput) {
+    emailInput.addEventListener("focusout", checkEmail);
+}
+
+// 비밀번호 검사
+function checkPassword() {
+    const passwordValue = passwordInput.value;
+
+    if (!passwordValue) {
+        showError(passwordInput, "password-empty");
+    } else if (passwordValue.length < 8) {
+        hideError(passwordInput, "password-empty");
+        showError(passwordInput, "password-invalidate");
+    } else {
+        isPasswordValidate = true;
+        hideError(passwordInput, "password-empty");
+        hideError(passwordInput, "password-invalidate");
+    }
+    submitButtonInvalidate();
+}
+
+if (passwordInput) {
+    passwordInput.addEventListener("focusout", checkPassword);
+}
+
+// 닉네임 검사
+function checkName() {
+    const nameValue = nameInput.value;
+    if (!nameValue) {
+        showError(nameInput, "name-empty");
+        isNameValidate = false;
+    } else {
+        isNameValidate = true;
+        hideError(nameInput, "name-empty");
+    }
+    submitButtonInvalidate();
+}
+
+if (nameInput) {
+    nameInput.addEventListener("focusout", checkName);
+}
+
+// 비밀번호 확인
+function checkPasswordConfirm() {
+    const passwordValue = passwordInput.value;
+    const passwordConfirmValue = passwordConfirmInput.value;
+
+    if (!passwordConfirmValue) {
+        showError(passwordConfirmInput, "password-confirm-empty");
+        isPasswordConfirmValidate = false;
+    } else if (passwordValue !== passwordConfirmValue) {
+        hideError(passwordConfirmInput, "password-confirm-empty");
+        showError(passwordConfirmInput, "password-confirm-invalidate");
+        isPasswordConfirmValidate = false;
+    } else {
+        isPasswordConfirmValidate = true;
+        hideError(passwordConfirmInput, "password-confirm-empty");
+        hideError(passwordConfirmInput, "password-confirm-invalidate");
+    }
+    submitButtonInvalidate();
+}
+
+if (passwordConfirmInput) {
+    passwordConfirmInput.addEventListener("focusout", checkPasswordConfirm);
+}
+
+// 조건 모두 충족 시 로그인, 회원가입 버튼 활성화
+function submitButtonInvalidate() {
+    if (signupForm) {
+        if (isEmailValidate && isPasswordValidate && isPasswordConfirmValidate && isNameValidate) {
+            signupButton.disabled = false;
+            window.location.href="/login.html";
+        } else {
+            signupButton.disabled = true;
+        }
+    } else if (loginForm) {
+        if (isEmailValidate && isPasswordValidate) {
+            loginButton.disabled = false;
+            window.location.href="/items.html";
+        } else {
+            loginButton.disabled = true;
+        }
+    }
+}
```

Comment:
P2:
중첩 if 문은 코드파악하기 어렵게 만듭니다.
가능하면 아래와 같이 너무 중첩되지 않게 작성해주세요.

```suggestion
function initSignupForm() {
    const isValid = isEmailValidate && isPasswordValidate && isPasswordConfirmValidate && isNameValidate;
 
    // 이코드가 잘 이해가 안됩니다. 버튼의 disabled설정이 바뀌는건 input 값이 변경될때마다 check해서 값이 valid하면 disabled를 변경해줘야하고, 페이지 이동을 버튼 click 이벤트가 발생했을때 이동되야하는거 아닌가 싶네요.   
    if (isValid) {
            signupButton.disabled = false;
            window.location.href="/login.html";
            return;
      }  
      signupButton.disabled = true;
}

function initLoginForm() {
    const isValid = isEmailValidate && isPasswordValidate
    if (isValid) {
        loginButton.disabled = false;
        window.location.href="/items.html";
        return;
    }     
   loginButton.disabled = true;
}

// 조건 모두 충족 시 로그인, 회원가입 버튼 활성화
function submitButtonInvalidate() {
    const loginForm = document.getElementById("login-form");
    const signupForm = document.getElementById("signup-form");

    if (signupForm) initSignupForm()
    if (loginForm) initLoginForm()
}
```